Sources/Routing/Parameter/Parameter.swift

```diff
@@ -32,15 +33,27 @@ public protocol Parameter {
     /// - returns: An instance of the `ResolvedParameter` type if one could be created.
     /// - throws: Throws an error if a `ResolvedParameter` could not be created.
     static func resolveParameter(_ parameter: String) throws -> ResolvedParameter
+    
+    /// Creates a `PathComponent` for this type which can be used
+    /// when registering routes to a router.
+    static var parameter: PathComponent { get }
 }
 
 // MARK: Methods
 
+extension Parameter where Self: LosslessDataConvertible {
+    public typealias ParameterType = Self
```

Comment:
This associated type indicates the expected type of parameter. It's needed to differentiate from the conforming type. For example

```swift
// matches /foo/10
// The conforming type (Likely `ResolvedParameter`) will be the same as `ParameterType`
"foo", Int.parameter

// matches /foo/1 where User.ID == Int
// The conforming type is different than the `ParameterType`
"foo", User.parameter
```